### PR TITLE
[docs] Configuration updates

### PR DESCRIPTION
**Pull request by bmorelli25**
My suggested updates based on the conversation here: https://github.com/elastic/apm-agent-java/pull/763/files#r309547992.

## Current docs:
Screen Shot 2019-08-06 at 11 26 02 AM

## This PR:
Screen Shot 2019-08-06 at 11 26 10 AM

### CODE REVIEW

**Comment by felixbarny** [comment]
Higher overrides lower May be misleading as the lower numbers in the numbered list actually override the higher numbers.

**Comment by bmorelli25** [comment]
Huh, I wasn't thinking about it in a number sense, just a high vs. low sense. Interesting, and a very good point! 

What about something like:
```asciidoc
The Agent gives precedence to the configuration sources as follows,
where Central configuration has the highest priority,
and `elasticapm.properties` has the lowest.
```

**Comment by felixbarny** [comment]
SGTM

File: docs/configuration.asciidoc [hunk 1]
```diff
@@ -5,21 +5,21 @@ Please only make changes in configuration.asciidoc.ftl
...
 [[configuration]]
 == Configuration
-To adapt the Elastic APM agent to your needs,
-you can configure it using different configuration sources,
+You can configure the Elastic APM Agent using different configuration sources,
 which have different naming conventions for the property key.
-The first configuration sources override the configuration values of over the latter sources.
+The Agent gives precedence to the configuration sources as follows,
+where Central configuration has the highest priority,
+and `elasticapm.properties` has the lowest.
```
Comment (eyalkoren) [inline docs/configuration.asciidoc]:
```suggestion
and `elasticapm.properties` has the lowest:
```

**Comment by bmorelli25** [comment]
Test failure looks unrelated, but I'll leave this for one of you two to merge.

**Pull request merged**